plugins/kubernetes-backend/src/kubernetes-auth-translator/AzureIdentityKubernetesAuthTranslator.ts

```diff
@@ -16,13 +16,24 @@
 
 import { KubernetesAuthTranslator } from './types';
 import { AzureClusterDetails } from '../types/types';
-import { DefaultAzureCredential } from '@azure/identity';
+import {
+  AccessToken,
+  DefaultAzureCredential,
+  TokenCredential,
+} from '@azure/identity';
 
 const aksScope = '6dae42f8-4368-4678-94ff-3960e28e3630/.default'; // This scope is the same for all Azure Managed Kubernetes
 
 export class AzureIdentityKubernetesAuthTranslator
   implements KubernetesAuthTranslator
 {
+  private tokenCredential: TokenCredential;
+  private accessToken: AccessToken | null = null;
+
+  constructor(tokenCredential?: TokenCredential) {
+    this.tokenCredential = tokenCredential || new DefaultAzureCredential();
+  }
```

Comment:
```suggestion
  constructor(private readonly tokenCredential: TokenCredential = new DefaultAzureCredential) { }
```

Another way of doing this :grin: